### DISCUSSION TITLE
Support reversing transaction order on import

### DISCUSSION THREAD

**Discussion by f-izzo** [ananthakumaran/paisa#105] [Ideas]
I have written a template for the transaction file from my bank: [CheBanca!](https://www.chebanca.it/)
I realized too late that the file provided by my bank has newest transactions at the top, while ledger expects oldest transaction at the top.
Using the original order (newest at the top) breaks ledger [Balance Assertions](https://ledger-cli.org/doc/ledger3.html#Balance-assertions).

Is it possible to provide a helper to parse the transaction file from bottom to top or to reverse the resulting transaction order?

Anonymized report example is attached
[CheBanca_example.xlsx](https://github.com/ananthakumaran/paisa/files/13545987/CheBanca_example.xlsx)

Current template: "CheBanca! Lista Movimenti"
```
{{#if (isDate ROW.A "DD/MM/YYYY")}}
  {{date ROW.A "DD/MM/YYYY"}} {{ROW.C}}
  {{#if (isBlank ROW.D) }}
    {{predictAccount prefix="Expenses"}}      {{negate (trim ROW.E)}} EUR
    Assets:Checking
  {{else}}
    Assets:Checking                           {{trim ROW.D}} EUR
    {{predictAccount prefix="Income"}}
  {{/if}}
{{/if}}
```

**Comment by ananthakumaran** [comment]
Let me think about this a bit. Implementing the reverse is very easy, I need to think about where to expose this option. Adding helper won't work because the template is called for each row.

**Reply by ananthakumaran** [reply]
@Nimayer I have added the feature to master https://github.com/ananthakumaran/paisa/commit/72a9ded5062c284cdc63f528ffea59d59d2bc633, you can checkout the demo page https://demo.paisa.fyi/ledger/import. Will be available on the next release.

**Comment by f-izzo** [comment]
I was also able to automatically add a Balance Assertion on import using this updated template.
```
{{#if (isDate ROW.A "DD/MM/YYYY")}}
  {{date ROW.A "DD/MM/YYYY"}} {{ROW.C}}
  {{#if (isBlank ROW.D) }}
    {{predictAccount prefix="Expenses"}}      {{negate (trim ROW.E)}} EUR
    Assets:Checking
  {{else}}
    Assets:Checking                           {{trim ROW.D}} EUR
    {{predictAccount prefix="Income"}}
  {{/if}}
{{/if}}
{{#if (eq ROW.A "Saldo contabile:")}}
  {{date (findBelow A regexp="PERIODO: dal .* fino al (.*)" group=1) "DD/MM/YYYY"}}
  Assets:Checking                                    0 EUR = {{trim ROW.B}} EUR
{{/if}}
```

However, since the balance is placed at the top of the spreadsheet, I can only put the assertion at the top of the ledger, while it should go at the bottom.

Maybe you can find a way to solve both issues at once.

Thank you in advance!

**Comment by f-izzo** [comment]
The new feature works perfectly! :grin: 

It even handles the Balance Assertion that I mentioned in the comment above.

Thank you very much for the quick implementation!
